compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/type/builders/BallerinaTupleTypeBuilder.java

```diff
@@ -68,16 +70,25 @@ public TypeBuilder.TUPLE withRestType(TypeSymbol restType) {
 
     @Override
     public TupleTypeSymbol build() {
-        List<BType> memberBTypes = new ArrayList<>();
-        for (TypeSymbol memberType : memberTypes) {
-            memberBTypes.add(getMemberType(memberType));
+        List<BTupleMember> memberTypes = new ArrayList<>();
+        for (TypeSymbol memberType : this.memberTypes) {
+            BType type = getMemberType(memberType);
+            BVarSymbol varSymbol;
+            if (type == symTable.noType) {
```

Comment:
What kind of scenario is this true for?

Reply:
I think this is used when there is an undefined tuple member

Reply:
Example please?

Reply:
`noType` will be considered for a tuple like `[1, x]` where `x` is undefined but that won't happen in this particular test case so I guess we can safely remove it. Not sure why it was needed originally, will remove for now and check and update.